### PR TITLE
5: Remove pipeline support from process

### PR DESCRIPTION
**Pull request by jeyvison**
Hey Folks :) Is that what you meant with removing pipeline support? If it's not, let me know. Also, I think something must be changed in ProcessBuilder, am i right? And this PR has a test problem i'm not able to fix but if any of you give me some directions i can fix it and resend :)

Progress
--------
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

Approvers
---------
 * Robin Westberg ([rwestberg](@rwestberg) - **Reviewer**)
 * Erik Helin ([ehelin](@edvbld) - **Reviewer**)

### CODE REVIEW

**Comment by rwestberg** [comment]
> Hey Folks :) Is that what you meant with removing pipeline support? If it's not, let me know. 

Hi! Thanks for the PR! Yes, this is indeed what we meant. (There's actually even a bit more complexity that can be removed, I'll add a more detailed review).

> Also, I think something must be changed in ProcessBuilder, am i right?

Do you mean java.lang.ProcessBuilder? That should not need any changes, perhaps I'm misunderstanding the question..

> And this PR has a test problem i'm not able to fix but if any of you give me some directions i can fix it and resend :)

All tests pass when I run them in your branch, what kind of problem did you see?

File: process/src/main/java/org/openjdk/skara/process/Execution.java [hunk 1]
```diff
@@ -253,11 +206,7 @@ private Result createResult() {
         finished = false;
 
         try {
-            if (processBuilders.size() == 1) {
```
Comment (rwestberg) [inline process/src/main/java/org/openjdk/skara/process/Execution.java]: And without pipes, processBuilders don't even need to be a list.

File: process/src/main/java/org/openjdk/skara/process/Execution.java [hunk 2]
```diff
@@ -138,52 +137,6 @@ private void prepareRedirects() throws IOException {
...
-                }
-                throw new InterruptedException("Command '" + cmd + "' didn't finish in " + timeout + ", terminated");
-            }
-            remainingTimeout = remainingTimeout.minus(Duration.between(start, Instant.now()));
-            start = Instant.now();
-        }
-    }
-
     private void startProcess() throws IOException {
         cmd = String.join(" ", getLastProcessBuilder().command());
```
Comment (rwestberg) [inline process/src/main/java/org/openjdk/skara/process/Execution.java]: The getLastProcessBuilder method could be removed as there is never more than one now.

**Comment by jeyvison** [comment]
@rwestberg Done :) Hope it's better now.

File: process/src/main/java/org/openjdk/skara/process/Execution.java [hunk 3]
```diff
@@ -107,92 +105,39 @@ public int status() {
...
-        cmd = String.join(" ", getLastProcessBuilder().command());
+        cmd = String.join(" ", processBuilder.command());
         log.fine("Executing '" + cmd + "'");
 
         prepareRedirects();
-        start = Instant.now();
 
         processes = new LinkedList<>();
-        processes.add(getLastProcessBuilder().start());
+        processes.add(processBuilder.start());
```
Comment (edvbld) [inline process/src/main/java/org/openjdk/skara/process/Execution.java]: We can also remove the field `private List<java.lang.Process> processes` now, that can just be turned into `private java.lang.Process process` (the list was only there to support pipelines).

File: process/src/main/java/org/openjdk/skara/process/Execution.java [hunk 4]
```diff
@@ -239,25 +184,20 @@ private Result createResult() {
         }
 
 
-        var command = processBuilders.stream()
-                                     .map(pb -> String.join(" ", pb.command()))
-                                     .reduce("", (res, cmd) -> res.isEmpty() ? cmd : res + " | " + cmd);
+        var command = String.join(" ", processBuilder.command());
```
Comment (edvbld) [inline process/src/main/java/org/openjdk/skara/process/Execution.java]: Can't you just the field `cmd` here instead that you updated on line 134?

**Comment by jeyvison** [comment]
@edvbld Third time is the luck one. Done :)

**Comment by jeyvison** [comment]
Should https://github.com/openjdk/skara/blob/d76aaa2ff9e8809833137c899ec819977d5903a4/process/src/main/java/org/openjdk/skara/process/Process.java#L40 be just a single command too?

**Comment by rwestberg** [comment]
Btw, I changed the title of the PR to conform to our preferred format (this will be automated in the future).

**Comment by rwestberg** [comment]
> Should
> 
> https://github.com/openjdk/skara/blob/d76aaa2ff9e8809833137c899ec819977d5903a4/process/src/main/java/org/openjdk/skara/process/Process.java#L40
> 
> be just a single command too?

And no, that one is fine, commands (executable and its arguments) are passed to ProcessBuilder as a list: https://github.com/openjdk/skara/blob/d76aaa2ff9e8809833137c899ec819977d5903a4/process/src/main/java/org/openjdk/skara/process/Process.java#L87

**Pull request closed without merging**